src/NHibernate.Test/NHSpecificTest/NH3985/Mappings.hbm.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<hibernate-mapping xmlns="urn:nhibernate-mapping-2.2" assembly="NHibernate.Test" namespace="NHibernate.Test.NHSpecificTest.NH3985">
+	<class name="Process" table="Processes">
+		<id name="ProcessID" type="Int32">
+			<column name="ProcessID" not-null="true" sql-type="INT" />
+			<generator class="identity" />
```

Comment:
Identity with Oracle use decimal. Using `native` as generator would be safer by the way, current NHibernate Oracle dialect does not implement identity though Oracle supports it. I believe it does fallback to sequence nether-less, so will work like `native`, but at least for `native` that is its documented behavior.

You may also consider using an app side generator like `guid.comb`, that helps avoid db specific issues.